pmd-cataloger/src/main/java/sfdc/sfdx/scanner/pmd/XmlFileFinder.java

```diff
@@ -34,39 +35,66 @@
 		}
 	}
 
+	/**
+	 * Represents a container for XML. This can either be a Jar file which may contain multiple XML files or an XML
+	 * file that contains itself.
+	 */
+	public static final class XmlContainer {
```

Comment:
Added this abstraction to provide enough information when findXmlInPath is invoked with a directory. This allows the caller to inform the user which jar has the conflicting path.